packages/ui/src/components/va-modal/VaModal.vue

```diff
@@ -82,195 +105,160 @@
-})
-export default class VaModal extends mixins(
-  ColorMixin,
-  StatefulMixin,
-  ModalPropsMixin,
-) {
-  created () {
-    watch(() => this.valueComputed, (valueComputed: boolean) => {
-      if (valueComputed) {
-        window.addEventListener('keyup', this.listenKeyUp)
-      } else {
-        window.removeEventListener('keyup', this.listenKeyUp)
-      }
+  emits: [
+    ...useStatefulEmits,
+    'update:modelValue', 'cancel', 'ok', 'before-open', 'open', 'before-close', 'close', 'click-outside',
+  ],
+  props: {
+    ...useStatefulProps,
+    modelValue: { type: Boolean as PropType<boolean>, default: false },
+    attachElement: { type: String as PropType<string>, default: 'body' },
+    disableAttachment: { type: Boolean as PropType<boolean>, default: false },
+    title: { type: String as PropType<string>, default: '' },
+    message: { type: String as PropType<string>, default: '' },
+    okText: { type: String as PropType<string>, default: 'OK' },
+    cancelText: { type: String as PropType<string>, default: 'Cancel' },
+    hideDefaultActions: { type: Boolean as PropType<boolean>, default: false },
+    fullscreen: { type: Boolean as PropType<boolean>, default: false },
+    mobileFullscreen: { type: Boolean as PropType<boolean>, default: true },
+    noDismiss: { type: Boolean as PropType<boolean>, default: false },
+    noOutsideDismiss: { type: Boolean as PropType<boolean>, default: false },
+    noEscDismiss: { type: Boolean as PropType<boolean>, default: false },
+    maxWidth: { type: String as PropType<string>, default: '' },
+    maxHeight: { type: String as PropType<string>, default: '' },
+    size: {
+      type: String as PropType<'medium' | 'small' | 'large'>,
+      default: 'medium',
+      validator: (size: string) => ['medium', 'small', 'large'].includes(size),
+    },
+    fixedLayout: { type: Boolean as PropType<boolean>, default: false },
+    withoutTransitions: { type: Boolean as PropType<boolean>, default: false },
+    overlay: { type: Boolean as PropType<boolean>, default: true },
+    overlayOpacity: { type: [Number, String] as PropType<number | string>, default: 0.6 },
+    zIndex: { type: [Number, String] as PropType<number | string | undefined>, default: undefined },
+  },
+  setup (props, { emit, slots }) {
+    const rootElement = ref()
+    const { valueComputed } = useStateful(props, emit)
+
+    const computedClass = computed(() => ({
+      'va-modal--fullscreen': props.fullscreen,
+      'va-modal--mobile-fullscreen': props.mobileFullscreen,
+      'va-modal--fixed-layout': props.fixedLayout,
+      [`va-modal--size-${props.size}`]: props.size !== 'medium',
+    }))
+    const computedModalContainerStyle = computed(() => ({ 'z-index': props.zIndex } as StyleValue))
+    const computedOverlayStyles = computed(() => {
+      // NOTE Not sure exactly what that does.
+      // Supposedly solves some case when background wasn't shown.
+      // As a side effect removes background from nested modals.
```

Comment:
Pretty sure we wanted to prevent background from getting darker and darker as more modals were being added.